File: apollo-integration/src/test/java/com/apollographql/android/impl/NoFileSystem.java
package com.apollographql.android.impl;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.IOException;

import okhttp3.internal.io.FileSystem;
import okio.Sink;
import okio.Source;

class NoFileSystem implements FileSystem {
  @Override public Source source(File file) throws FileNotFoundException {
    throw new UnsupportedOperationException();
  }

  @Override public Sink sink(File file) throws FileNotFoundException {
    throw new UnsupportedOperationException();
  }

  @Override public Sink appendingSink(File file) throws FileNotFoundException {
    throw new UnsupportedOperationException();
  }

  @Override public void delete(File file) throws IOException {
    throw new UnsupportedOperationException();
  }

  @Override public boolean exists(File file) {
    throw new UnsupportedOperationException();
  }

  @Override public long size(File file) {
    throw new UnsupportedOperationException();
  }

  @Override public void rename(File from, File to) throws IOException {
    throw new UnsupportedOperationException();
  }

  @Override public void deleteContents(File directory) throws IOException {
    throw new UnsupportedOperationException();
  }
}

File: apollo-integration/src/test/kotlin/com/apollographql/apollo/NoFileSystem.kt
package com.apollographql.apollo

import com.apollographql.apollo.cache.http.internal.FileSystem
import okio.Sink
import okio.Source
import java.io.File
import java.io.FileNotFoundException
import java.io.IOException

internal class NoFileSystem : FileSystem {
  @Throws(FileNotFoundException::class)
  override fun source(file: File): Source {
    throw UnsupportedOperationException()
  }

  @Throws(FileNotFoundException::class)
  override fun sink(file: File): Sink {
    throw UnsupportedOperationException()
  }

  @Throws(FileNotFoundException::class)
  override fun appendingSink(file: File): Sink {
    throw UnsupportedOperationException()
  }

  @Throws(IOException::class)
  override fun delete(file: File) {
    throw UnsupportedOperationException()
  }

  override fun exists(file: File): Boolean {
    throw UnsupportedOperationException()
  }

  override fun size(file: File): Long {
    throw UnsupportedOperationException()
  }

  @Throws(IOException::class)
  override fun rename(from: File, to: File) {
    throw UnsupportedOperationException()
  }

  @Throws(IOException::class)
  override fun deleteContents(directory: File) {
    throw UnsupportedOperationException()
  }
}